### PR TITLE
[GLT-3436][GLT-3271] Allow wildcard and null searches

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/util/DbUtils.java [hunk 1]
```diff
@@ -41,44 +48,108 @@
...
+    String sanitized = sanitizeQueryString(query.getText());
+    MatchMode matchMode = getMatchMode(query);
+    if (searchProperties.length == 1) {
+      return textRestriction(searchProperties[0], sanitized, matchMode);
+    } else {
+      Criterion[] criteria = new Criterion[searchProperties.length];
+      for (int i = 0; i < searchProperties.length; i++) {
+        criteria[i] = textRestriction(searchProperties[i], sanitized, matchMode);
+      }
+      if (sanitized == null) {
```
Comment (apmasell) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/util/DbUtils.java:105]: This is slightly confusing...if the query is `IS NULL` and we have multiple properties, then it becomes _all are null_, but if we're searching for text, then _any field has this text_?

**Review by apmasell — APPROVED** [review]

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/util/DbUtils.java (hunk 1 above)
Comment (djcooke) [reply sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/util/DbUtils.java:105]: For searches that check multiple fields, if you're looking for nulls, it should be null in all those fields. e.g. "external" checks both `externalName` and `secondaryIdentifier`. Searching "external:" finds items with neither specified this way. If it were "any is null" then the query would always return everything.

Comment (apmasell) [reply sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/util/DbUtils.java:105]: Makes sense. Can you stick this in a comment?

Comment (djcooke) [reply sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/util/DbUtils.java:118]: done

**Review by avarsava — APPROVED** [review]

**Pull request merged**